src/odemis/driver/orsay.py

```diff
@@ -0,0 +1,766 @@
+        else:
+            self._pneumaticSuspension = pneumaticSuspension(parent=self, daemon=daemon, **kwargs)
+            self.children.value.add(self._pneumaticSuspension)
+
+        # create the pressure child for the chamber
+        try:
+            kwargs = children["pressure"]
+        except (KeyError, TypeError):
+            logging.info("Orsay was not given a 'pressure' child")
+        else:
+            self._pressure = vacuumChamber(parent=self, daemon=daemon, **kwargs)
+            self.children.value.add(self._pressure)
+
+        # create the pumping system child
+        try:
+            kwargs = children["pumping-system"]
+        except (KeyError, TypeError):
+            logging.info("Orsay was not given a 'pumping-system' child")
+        else:
+            self._pumpingSystem = pumpingSystem(parent=self, daemon=daemon, **kwargs)
+            self.children.value.add(self._pumpingSystem)
+
+        # create the UPS child
+        try:
+            kwargs = children["ups"]
+        except (KeyError, TypeError):
+            logging.info("Orsay was not given a 'ups' child")
+        else:
+            self._ups = UPS(parent=self, daemon=daemon, **kwargs)
+            self.children.value.add(self._ups)
+
+        self.connected.set()
+
+    def on_connect(self):
+        """
+        Defines the shorthand access points and connects parameter callbacks for the Orsay server.
+        Needs to be called after connection and reconnection to the server.
+        """
+        self.datamodel = self._device.datamodel
+        self.datamodel.HybridPlatform.ProcessInfo.Subscribe(self._updateProcessInfo)
+
+        self.update_VAs()
+
+    def update_VAs(self):
+        """
+        Update the VA's. Should be called after reconnection to the server
+        """
+        self._updateProcessInfo()
+
+    def _check_connection_thread(self):
+        """
+        Once in a while, check the connection to the Orsay server, reconnect if needed and update all VA's
+        """
+        while not self._stop_connection_check.is_set():
+            if self._device.HttpConnection._HTTPConnection__response is None or \
+                    self._device.MessageConnection.Connection._HTTPConnection__response is None:
+                self.connected.clear()  # report the connection is broken
+                self._device.HttpConnection.close()  # close the current connection
+                self._device.MessageConnection.Connection.close()
+                self._device = None  # destroy the current connection object
+                try:
+                    self._device = Connection(self._host)
+                    time.sleep(1)
+                    self.on_connect()
+                    for child in self.children.value:
+                        child.on_connect()
+                    self.connected.set()  # report the connection is fixed
+                except Exception:
+                    logging.warning("Trying to reconnect to Orsay server")
```

Comment:
The exception I'm trying to catch here is the one raised by Connection(self._host) in case the connection is still not there. The Orsay API, however, only raises a simple exception in that case, no specific type, no message attached to it. I'll see what I can do to change their API so it raises proper exceptions and make another merge request to Orsay's GitLab repo.